packages/lodestar/src/sync/utils/assertPeerRelevance.ts

```diff
@@ -0,0 +1,110 @@
+import {computeStartSlotAtEpoch, getBlockRootAtSlot} from "@chainsafe/lodestar-beacon-state-transition";
+import {IBeaconConfig} from "@chainsafe/lodestar-config";
+import {Epoch, ForkDigest, Root, Status} from "@chainsafe/lodestar-types";
+import {LodestarError} from "@chainsafe/lodestar-utils";
+import {toHexString} from "@chainsafe/ssz";
+import {IBeaconChain} from "../../chain";
+import {GENESIS_EPOCH} from "../../constants";
+
+// TODO: Why this value? (From Lighthouse)
+const FUTURE_SLOT_TOLERANCE = 1;
+
+export enum IrrelevantPeerErrorCode {
+  INCOMPATIBLE_FORKS = "IRRELEVANT_PEER_INCOMPATIBLE_FORKS",
+  DIFFERENT_CLOCKS = "IRRELEVANT_PEER_DIFFERENT_CLOCKS",
+  GENESIS_NONZERO = "IRRELEVANT_PEER_GENESIS_NONZERO",
+  DIFFERENT_FINALIZED = "IRRELEVANT_PEER_DIFFERENT_FINALIZED",
+}
+
+type IrrelevantPeerErrorType =
+  | {code: IrrelevantPeerErrorCode.INCOMPATIBLE_FORKS; ours: ForkDigest; theirs: ForkDigest}
+  | {code: IrrelevantPeerErrorCode.DIFFERENT_CLOCKS; slotDiff: number}
+  | {code: IrrelevantPeerErrorCode.GENESIS_NONZERO; root: string}
+  | {code: IrrelevantPeerErrorCode.DIFFERENT_FINALIZED; expectedRoot: string; remoteRoot: string};
+
+export class IrrelevantPeerError extends LodestarError<IrrelevantPeerErrorType> {}
+
+/**
+ * Process a `Status` message to determine if a peer is relevant to us. If the peer is
+ * irrelevant the reason is returned.
+ */
+export async function assertPeerRelevance(remote: Status, chain: IBeaconChain, config: IBeaconConfig): Promise<void> {
```

Comment:
just an aside, i think `IBeaconChain` should publicly expose a `IBeaconChain#config`

Reply:
That's a good idea!